File: app/src/main/java/com/example/weizhenbin/wangebug/base/UrlTypeEnum.java
package com.example.weizhenbin.wangebug.base;

import java.io.Serializable;

/**
 * Created by weizhenbin on 2018/10/10.
 */
public enum UrlTypeEnum implements Serializable{
    unknown(), code(1),news(2);

    private int typeValue;

     UrlTypeEnum(int typeValue) {
      this.typeValue=typeValue;
    }
    UrlTypeEnum() {
    }
    public int getTypeValue() {
        return typeValue;
    }

    public static UrlTypeEnum valueOf(int value) {
        switch (value) {
            case 1:
                return UrlTypeEnum.code;
            case 2:
                return UrlTypeEnum.news;
            default:
                return UrlTypeEnum.unknown;
        }
    }



}

File: app/src/main/java/com/example/weizhenbin/wangebug/base/UrlTypeEnum.kt
package com.example.weizhenbin.wangebug.base

import java.io.Serializable

/**
 * Created by weizhenbin on 2018/10/10.
 */
enum class UrlTypeEnum(val typeValue:Int) : Serializable {
     UNKNOWN(-1), CODE(1), NEWS(2);

     companion object {
          fun getValueFromType(typeValue: Int):UrlTypeEnum{
               return when(typeValue){
                    1->UrlTypeEnum.CODE
                    2->UrlTypeEnum.NEWS
                    else -> UrlTypeEnum.UNKNOWN
               }
          }
     }
}


